src/frontend/screens/CategoryChooser.js

```diff
@@ -71,20 +71,52 @@ const Item = React.memo(
 const CategoryChooser = ({ navigation }: { navigation: NavigationProp }) => {
   const [{ presets }] = useContext(ConfigContext);
   const [{ value: draftValue }, { updateDraft }] = useDraftObservation();
-
   const presetsList = Array.from(presets.values())
     // Sort presets by sort property and then by name, then filter only point presets
     .sort(presetCompare)
     // Only show presets where the geometry property includes "point"
     .filter(p => p.geometry.includes("point"));
-
   const handleSelectPreset = (preset: Preset) => {
-    updateDraft({
-      tags: {
-        ...(draftValue || {}).tags,
-        categoryId: preset.id,
-      },
-    });
+    const newTags = preset.tags;
+    if (Object.keys(draftValue.tags).length > 0) {
+      // We're updating an observation
+      // Get tags from old category
+      const oldTags = draftValue.tags;
+      const oldCategoryTags = presetsList.filter(
+        p => p.id === oldTags.categoryId
+      )[0].tags;
+      // Get list of new tags
+      const updatedTags = Object.keys(oldTags).reduce((previous, current) => {
+        // Check if tag belongs to old category
+        const hasOldInfo =
+          Object.entries(oldCategoryTags).filter(([key, value]) => {
+            return key === current && value === oldTags[current];
+          }).length > 0;
```

Comment:
```suggestion
        const tagIsFromPreset = typeof oldCategoryTags[current] !== 'undefined' && oldCategoryTags[current] === oldTags[current]
```
I think this is easier to read and understand what we are checking.